skiko/src/jvmMain/cpp/windows/directXRedrawer.cc

```diff
@@ -220,40 +225,75 @@ extern "C"
         return false;
     }
 
-    bool defineHardwareAdapter(IDXGIFactory4 *pFactory, IDXGIAdapter1 **ppAdapter)
+    bool defineHardwareAdapter(DXGI_GPU_PREFERENCE adapterPriority, IDXGIFactory4 *pFactory, IDXGIAdapter1 **ppAdapter)
     {
         *ppAdapter = nullptr;
-        for (UINT adapterIndex = 0;; ++adapterIndex)
+
+#if defined(__dxgi1_6_h__) && defined(NTDDI_WIN10_RS4)
+        gr_cp<IDXGIFactory6> factory6;
+        if (SUCCEEDED(pFactory->QueryInterface(IID_PPV_ARGS(&factory6))))
         {
-            IDXGIAdapter1 *pAdapter = nullptr;
-            if (DXGI_ERROR_NOT_FOUND == pFactory->EnumAdapters1(adapterIndex, &pAdapter))
+            for (UINT adapterIndex = 0;; ++adapterIndex)
             {
-                break;
+                IDXGIAdapter1 *pAdapter = nullptr;
+                if (!SUCCEEDED(factory6->EnumAdapterByGpuPreference(adapterIndex, adapterPriority, IID_PPV_ARGS(&pAdapter))))
+                {
+                    break;
+                }
+
+                DXGI_ADAPTER_DESC1 desc;
+                pAdapter->GetDesc1(&desc);
+                std::wstring currentAdapterName(desc.Description);
+                fwprintf(stderr, L"Adapter #%d: %s  %d\n", adapterIndex, currentAdapterName.c_str(), adapterPriority);
```

Comment:
We shall not print anything by default, please remove the printf.

Reply:
Sure. Done.